cmd/juju/cloud/add.go

```diff
@@ -56,11 +64,15 @@ type addCloudCommand struct {
 
 	// CloudFile is the name of the cloud YAML file.
 	CloudFile string
+
+	cloudQueryer CloudQueryer
 }
 
 // NewAddCloudCommand returns a command to add cloud information.
-func NewAddCloudCommand() cmd.Command {
-	return &addCloudCommand{}
+func NewAddCloudCommand(cloudQueryer CloudQueryer) cmd.Command {
+	return &addCloudCommand{
+		cloudQueryer: cloudQueryer,
```

Comment:
Tests are code too, and tests that are difficult to write are a design smell. See my other comment about reusing well-tested modules across multiple commands.